solutions/src/main/java/ru/sberbank/school/task02/Converter.java

```diff
@@ -0,0 +1,63 @@
+package ru.sberbank.school.task02;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import ru.sberbank.school.task02.exception.FxConversionException;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import ru.sberbank.school.task02.util.Volume;
+
+public class Converter implements FxConversionService {
+    private ExternalQuotesService externalQuotesService;
+
+    public Converter(ExternalQuotesService externalQuotesService) {
+        this.externalQuotesService = externalQuotesService;
+    }
+
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+        if (operation == null || symbol == null || amount == null) {
+            throw new NullPointerException();
+        }
+
+        if (amount.equals(BigDecimal.ZERO)) {
+            throw new IllegalArgumentException("Введите корректное значение объема!");
+        }
+
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+
+        if (quotes.isEmpty()) {
+            throw new FxConversionException("Список валют пуст!");
+        }
+
+        final Quote result = getRightQuota(quotes,amount);
+
+        boolean isSell = operation == ClientOperation.SELL;
+        return isSell ? result.getBid() : result.getOffer();
+    }
+
+    private static Quote getRightQuota(List<Quote> quotes, BigDecimal amount) {
+        Quote result = null;
+
+        for (Quote q : quotes) {
+            BigDecimal quoteVolumeSize = q.getVolumeSize();
+            if (amount.compareTo(quoteVolumeSize) < 0) {
+                if (result == null || result.getVolumeSize().compareTo(quoteVolumeSize) > 0) {
+                    result = q;
+                }
+            }
+        }
+
+        if (result == null) {
+            result = quotes.get(0);
+        }
+
+        return result;
+
+  }
```

Comment:
&#39;method def rcurly&#39; has incorrect indentation level 2, expected level should be 4.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-143779071>*